src/background/main.js

```diff
@@ -386,6 +394,27 @@ class Main {
     }
   }
 
+  async betaDecisionRequired() {
+    // Compute proxy state does the magic.
+    if (await this.computeProxyState()) {
+      await this.ui.update();
+    }
+  }
+
+  async betaDecisionMade() {
+    await this.fxa.refreshProfileData();
+
+    // Compute proxy state does the magic.
```

Comment:
Make a common helper function between this and the contents of `betaDecisionRequired`?